src/credentialstore/credentialstore.ts

```diff
@@ -26,44 +44,141 @@ export class CredentialStore implements ICredentialStore {
      * @param {string} credentialId the ID uniquely identifying this credential
      * @returns {Promise<Credential>} Promise that resolved to the credential, or undefined if not found
      */
-    public readCredential(credentialId: string): Promise<Contracts.Credential> {
-        let self = this;
-        let cred: Contracts.Credential = new Contracts.Credential();
-        cred.credentialId = credentialId;
-        return new Promise<Contracts.Credential>( (resolve, reject) => {
-            self._client
-            .sendRequest(Contracts.ReadCredentialRequest.type, cred)
-            .then(returnedCred => {
-                resolve(returnedCred);
-            }, err => reject(err));
-        });
+    public async readCredential(credentialId: string): Promise<Contracts.Credential> {
+        const cred: Contracts.Credential = new Contracts.Credential();
+        try {
+            let returnedCred: Contracts.Credential;
+
+            // Use native credential if the setting is on
+            if (this._useNativeCredentials) {
+                returnedCred = await this.nativeReadCredential(credentialId, cred);
```

Comment:
I'd just return the calls directly instead of awaiting - that'll be doing the same thing and save having to unwrap and then re-wrap the promise. (here and in similar functions below)